schema_editor/app/scripts/userdata/user-service.js

```diff
@@ -42,27 +44,52 @@
 
         function getUser(userId) {
             var dfd = $q.defer();
-            var result = module.User.get({id: userId}, function () {
-                dfd.resolve(result);
+            module.User.get({id: userId}, function (user) {
+                if (user && user.groups) {
+                    // append attribute to response to indicate if user is an admin or not
+                    user.isAdmin = (user.groups.indexOf(ASEConfig.api.groups.admin) > -1) ? true : false;
+                }
+                dfd.resolve(user);
             });
             return dfd.promise;
         }
 
+        // Check whether user has write access
+        function canWriteRecords(userId, token) {
+            tmpToken = token;
```

Comment:
Before the user has been logged in to the frontend but after their user ID/token has been retrieved, their token gets used temporarily to authenticate a request for the rest of their user information, in order to determine what the frontend should show them access to, based on their group membership.